src/mlpack/core/util/cli_impl.hpp

```diff
@@ -23,16 +23,19 @@ namespace mlpack {
  * @tparam T The type of the parameter.
  * @param identifier The name of the parameter, eg foo in bar/foo.
  * @param description A string description of the parameter.
- * @param parent The name of the parent of the parameter,
- *   eg bar/foo in bar/foo/buzz.
- * @param required If required, the program will refuse to run
- *   unless the parameter is specified.
+ * @param parent The name of the parent of the parameter, e.g. bar/foo in
+ *     bar/foo/buzz.
+ * @param required If required, the program will refuse to run unless the
+ *     parameter is specified.
+ * @param input If true, the parameter is an input parameter (not an output
+ *     parameter).
  */
 template<typename T>
 void CLI::Add(const std::string& identifier,
               const std::string& description,
               const std::string& alias,
```

Comment:
I took a look in 7a34fe3; I think that the `parent` term is totally outdated and should actually be `alias`, since what is actually being passed there is the short option name (i.e. `-i` instead of `--input_file`).  Originally the idea was that the command-line interface would support arbitrarily hierarchical parameters, so each parameter needed to have some parent name associated with it.  But this idea ended up being trashed, so we only have a single level of parameters for the sake of simplicity.  Only these few lines of comments remained.